is_error_code_enum should not be defined when VULKAN_HPP_NO_EXCEPTIONS is defined

```
~/g/build (linux|✚22) $ gcc --version
gcc (Ubuntu 9.2.1-9ubuntu2) 9.2.1 20191008
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

~/g/build (linux|✚22) $ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 19.10
Release:	19.10
Codename:	eoan

```

I get this mysterious error whereas on Ubuntu 19.04 I did not:

```
/usr/include/c++/9/system_error:157:33: error: cannot convert ‘vk::Result’ to ‘std::errc’
  157 |       { *this = make_error_code(__e); }
      |                                 ^~~
      |                                 |
      |                                 vk::Result
/usr/include/c++/9/system_error:204:24: note:   initializing argument 1 of ‘std::error_code std::make_error_code(std::errc)’
  204 |   make_error_code(errc __e) noexcept
      |                   ~~~~~^~~
ninja: build stopped: subcommand failed.
```

Looking at the `system_error` file, I find this constructor for the `error_code` class:

```
    template<typename _ErrorCodeEnum, typename = typename
	     enable_if<is_error_code_enum<_ErrorCodeEnum>::value>::type>
      error_code(_ErrorCodeEnum __e) noexcept
      { *this = make_error_code(__e); }
```

Checking with `vulkan.hpp`, I find these lines:

```
namespace std
{
  template <>
  struct is_error_code_enum<VULKAN_HPP_NAMESPACE::Result> : public true_type
  {};
}
```

And indeed there is also a definition for `make_error_code` in `vulkan.hpp`:

```
  VULKAN_HPP_INLINE std::error_code make_error_code(Result e) VULKAN_HPP_NOEXCEPT
  {
    return std::error_code(static_cast<int>(e), errorCategory());
  }
```

**However**, `make_error_code` is wrapped in `VULKAN_HPP_NO_EXCEPTIONS`, whereas `is_error_code_enum` is not! This makes the constructor in `std::error_code` enabled, but there is no definition for `make_error_code`, giving me the error. I have `VULKAN_HPP_NO_EXCEPTIONS` defined.

--- a/VulkanHppGenerator.cpp
+++ b/VulkanHppGenerator.cpp
@@ -5212,12 +5212,14 @@ static const std::string constExpressionArrayCopy = R"(
 )";
 
   static const std::string is_error_code_enum = R"(
+#ifndef VULKAN_HPP_NO_EXCEPTIONS
 namespace std
 {
   template <>
   struct is_error_code_enum<VULKAN_HPP_NAMESPACE::Result> : public true_type
   {};
 }
+#endif
 )";
 
   static const std::string structResultValue = R"(

--- a/vulkan/vulkan.hpp
+++ b/vulkan/vulkan.hpp
@@ -11964,12 +11964,14 @@ namespace VULKAN_HPP_NAMESPACE
 #endif /*VK_USE_PLATFORM_XLIB_KHR*/
 } // namespace VULKAN_HPP_NAMESPACE
 
+#ifndef VULKAN_HPP_NO_EXCEPTIONS
 namespace std
 {
   template <>
   struct is_error_code_enum<VULKAN_HPP_NAMESPACE::Result> : public true_type
   {};
 }
+#endif
 
 namespace VULKAN_HPP_NAMESPACE
 {

